### ISSUE TITLE
[🐛] `snyk test` fails with Yarn v2 and Workspaces

### ISSUE THREAD

**Issue by mhassan1** [snyk/snyk#1322]
- `node -v`: 12.18.2
- `npm -v`: 6.14.5
- `snyk -v`: 1.373.0
- OS: OSX
- Command run: `snyk test --yarn-workspaces`

### Expected behaviour
Display Snyk results

### Actual behaviour
Throw error `yarn.lock parsing failed with an error: Unknown token: { line: 3, col: 2, type: 'INVALID', value: undefined } 3:2 in lockfile`

### Steps to reproduce
Create `package.json`:
```json
{
  "workspaces": [
    "packages/*"
  ],
  "private": true
}
```
Run:
```sh
yarn set version berry
yarn install
snyk test --yarn-workspaces
```

### Debug log
```text
error: InvalidUserInputError: yarn.lock parsing failed with an error: Unknown token: { line: 3, col: 2, type: 'INVALID', value: undefined } 3:2 in lockfile
      at YarnLockParser.parseLockFile (/Users/marc/.nvm/versions/node/v12.18.2/lib/node_modules/snyk/node_modules/snyk-nodejs-lockfile-parser/lib/parsers/yarn-lock-parse.ts:22:13)
      at Object.buildDepTree (/Users/marc/.nvm/versions/node/v12.18.2/lib/node_modules/snyk/node_modules/snyk-nodejs-lockfile-parser/lib/index.ts:84:45)
      at Object.processYarnWorkspaces [as handler] (/Users/marc/.nvm/versions/node/v12.18.2/lib/node_modules/snyk/src/lib/plugins/nodejs-plugin/yarn-workspaces-parser.ts:83:40)
      at Object.getDepsFromPlugin (/Users/marc/.nvm/versions/node/v12.18.2/lib/node_modules/snyk/src/lib/plugins/get-deps-from-plugin.ts:60:18)
      at assembleLocalPayloads (/Users/marc/.nvm/versions/node/v12.18.2/lib/node_modules/snyk/src/lib/snyk-test/run-test.ts:359:18)
      at runTest (/Users/marc/.nvm/versions/node/v12.18.2/lib/node_modules/snyk/src/lib/snyk-test/run-test.ts:151:22)
      at test (/Users/marc/.nvm/versions/node/v12.18.2/lib/node_modules/snyk/src/cli/commands/test/index.ts:119:13)
      at runCommand (/Users/marc/.nvm/versions/node/v12.18.2/lib/node_modules/snyk/src/cli/index.ts:49:25)
      at main (/Users/marc/.nvm/versions/node/v12.18.2/lib/node_modules/snyk/src/cli/index.ts:288:11)
```

**Comment by lili2311** [comment]
👋 @mhassan1 thanks for flagging this, yarn v2 is very fresh and doesn't yet support Yarn v2. We are discussing this issue internally as it is also tied to https://github.com/snyk/snyk/issues/1270

**Comment by mhassan1** [comment]
I've proposed a fix in #1336

**Comment by lili2311** [comment]
@mhassan1 thank you very much! We have some internal improvements to add to help make your PR smaller.

**Comment by lili2311** [comment]
@mhassan1 a version of CLI that supporrs Yarn 2 lockfiles + workspaces has been released, please confirm version `1.381.2` works as expected for you?

**Comment by mhassan1** [comment]
It's great, thank you!
